datafusion/functions-aggregate-common/src/aggregate/min_max/groups_accumulator_max_view.rs

```diff
@@ -0,0 +1,203 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+use arrow::array::{Array, ArrayRef, AsArray, BinaryViewBuilder, BooleanArray};
+use datafusion_common::{DataFusionError, Result};
+use datafusion_expr_common::groups_accumulator::{EmitTo, GroupsAccumulator};
+use std::sync::Arc;
+
+pub struct GroupsAccumulatorMaxStringView {
+    states: Vec<String>,
+}
+
+impl Default for GroupsAccumulatorMaxStringView {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl GroupsAccumulatorMaxStringView {
+    pub fn new() -> Self {
+        Self { states: Vec::new() }
+    }
+}
+
+impl GroupsAccumulator for GroupsAccumulatorMaxStringView {
+    fn update_batch(
+        &mut self,
+        values: &[ArrayRef],
+        group_indices: &[usize],
+        opt_filter: Option<&BooleanArray>,
+        total_num_groups: usize,
+    ) -> Result<()> {
+        if self.states.len() < total_num_groups {
+            self.states.resize(total_num_groups, String::new());
+        }
+
+        let input_array = &values[0];
+
+        for (i, &group_index) in group_indices.iter().enumerate() {
```

Comment:
I think much of the filter logic is handled by `accumulate_indices` https://github.com/apache/datafusion/blob/f54712d8c2545fc0dd60cdc693704992e9d82eef/datafusion/functions-aggregate/src/count.rs#L417-L424

You could likely avoid much of this repetition (and likely it would be faster) 

It woudl also be nice to avoid the duplication between min /max by using generics. Here is how the primitive one does it (passes in a comparison function)

https://github.com/apache/datafusion/blob/main/datafusion/functions-aggregate/src/min_max.rs#L119

Reply:
Okay so I would probably want to do something like this?

```rust
accumulate_indices(group_indices, input_array.logical_nulls().as_ref(), opt_filter, |group_index| {
        let value = input_array.as_binary_view().value(i);
    
        let value_str = std::str::from_utf8(value).map_err(|e| {
            DataFusionError::Execution(format!(
                "could not build utf8 from binary view {}",
                e
            ))
        }).unwrap();
    
        if self.states[group_index].is_empty() {
            self.states[group_index] = value_str.to_string();
        } else {
            let curr_value_bytes = self.states[group_index].as_bytes();
            if value < curr_value_bytes {
                self.states[group_index] = value_str.parse().unwrap();
            }
        } 
});
```

And then make this generic. I.E. I can pass a generic function in instead of:

```rust
        if self.states[group_index].is_empty() {
            self.states[group_index] = value_str.to_string();
        } else {
            let curr_value_bytes = self.states[group_index].as_bytes();
            if value < curr_value_bytes {
                self.states[group_index] = value_str.parse().unwrap();
            }
        } 
```

Afterwards I can likely use a const generic for down-casting here?

`let value = input_array.as_binary_view().value(i);`